### ISSUE TITLE
scale_fill_steps produces error when both breaks and labels are specified

### ISSUE THREAD

**Issue by ClaudioHeinrich** [tidyverse/ggplot2#4548]
When using `scale_fill_steps` or `scale_fill_steps2`, specifying two equal-length vectors as breaks and labels leads to an error.

```r
# some data:
random_data = data.frame(expand.grid(x = 1:10,y = 1:10),fill = runif(100,min = 0,max = 1))

# this works:
ggplot(random_data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps()

# this doesn't:
breaks = seq(0,1,length.out = 5)
labels = breaks
ggplot(random_data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps(breaks = breaks,labels = labels)
```

I am getting the error message 'Error: Breaks and labels are different lengths'.
I tried to find out which number of labels it expects, and ran this:
```{r}
for(i in 2:10)
{
  tryCatch({
    labels = seq(0,1,length.out = i)
    ggplot(random_data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps(breaks = breaks,labels = labels)
  }, error=function(e){print(i)})
}
```

This printed out the numbers from 2 to 10 excluding 5, which I don't understand because setting i to 5 and running the function passed to tryCatch manually does result in an error.

**Comment by ClaudioHeinrich** [comment]
This could be related to this recent bugfix: 
https://github.com/tidyverse/ggplot2/issues/4019

After this fix, the labels for the boundaries of the colorscale disappeared, so the number of default labels changed...

**Comment by ClaudioHeinrich** [comment]
Here you go:
``` r
library(ggplot2)
data = data.frame(expand.grid(x = 1:10,y = 1:10),fill = 1:100)

# this works:
ggplot(data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps()
```

![](https://i.imgur.com/rlErANg.png)

``` r

# this doesn't work, although it should:
breaks = seq(0,1,length.out = 5)
labels = breaks
ggplot(data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps(breaks = breaks,labels = labels)
#> Error: Breaks and labels are different lengths
```

![](https://i.imgur.com/I2siCUP.png)

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by ClaudioHeinrich** [comment]
Here you go:
``` r
library(ggplot2)
data = data.frame(expand.grid(x = 1:10,y = 1:10),fill = 1:100)

# this works:
ggplot(data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps()
```

![](https://i.imgur.com/Zu00gsM.png)

``` r

# this doesn't work, although it should (I think):
breaks = seq(0,100,length.out = 5)
labels = breaks
ggplot(data) + geom_tile(aes(x=x,y=y,fill = fill)) + scale_fill_steps(breaks = breaks,labels = labels)
#> Error: Breaks and labels are different lengths
```

![](https://i.imgur.com/eiN5v3b.png)

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by hadley** [comment]
Thanks. I've made it a little more minimal to make it easier to track down the problem:

``` r
library(ggplot2)
data <- data.frame(x = 1:3)

ggplot(data) +
  geom_col(aes(x, x, fill = x)) +
  scale_fill_steps(breaks = 1:2, labels = 1:2)
#> Error in `f()`:
#> ! Breaks and labels are different lengths
```

<sup>Created on 2022-03-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by yutannihilation** [comment]
It seems this is already fixed in the dev version (probably by #4849?).

``` r
library(ggplot2)
data <- data.frame(x = 1:3)

ggplot(data) +
  geom_col(aes(x, x, fill = x)) +
  scale_fill_steps(breaks = 1:2, labels = 1:2)
```

![](https://i.imgur.com/O5YuwsB.png)

<sup>Created on 2022-07-23 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
